### ISSUE TITLE
Multi-node service runs a scheduled job once

### ISSUE THREAD

**Issue by kanguki** [kanguki/goaid#6]
There are three replicas of service X running on different servers. Server X has to divide number N everyday at midnight once and only once. How to implement that?
I'm thinking of several approaches:
- Set one replica master to run the job: When schedule comes, each node clears the old master, waits a random amount of time, the first node done waiting will elect itself as the master, inform other. If it's ok from all other nodes (no other nodes are electing itself), then it will do the job, else another wait has to run again to reelect the master.  
- Set one replica master to run the job: When the schedule comes, each node waits a random amount of time, then the first waken up will go query table job, if time != schedule, it will update scheduled time for table job in db, and do it. The second comes in and see its job has been taken, it won't do anything
